frame/support/src/traits/tokens/fungible/conformance_tests/inspect_mutate.rs

```diff
@@ -0,0 +1,659 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2017-2023 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::traits::{
+	fungible::{Inspect, Mutate},
+	tokens::{
+		DepositConsequence, Fortitude, Precision, Preservation, Provenance, WithdrawConsequence,
+	},
+};
+use core::fmt::Debug;
+use sp_arithmetic::traits::AtLeast8BitUnsigned;
+
+//
+// mint_into
+//
+
+pub fn mint_into_success<T, AccountId, Balance>()
```

Comment:
I think we could also employ fuzzers - eventually.  
They look quite similar to your pattern so far and test the assumptions with many different inputs.  
A recent one is [this](https://github.com/paritytech/substrate/blob/344d3a6f850ef52e05368fdd58f7d9461658185b/primitives/arithmetic/fuzzer/src/per_thing_mult_fraction.rs#L42) and the `per_thing_from_rational` in that folder.

Reply:
Fuzzing is a great idea! I'll create an issue to work on that once some initial coverage is done.